### ISSUE TITLE
Feature Request: Anki Importing

### ISSUE THREAD

**Issue by Pcsl** [birtles/rikaichamp#44]
Other similar addons have Anki import but yours is at this moment my favorite one but it lacks of a main feature: Anki importing so you can mine vocabulary while reading a website.

Is there any plan to add this feature?

Thank you for your hard work!

**Comment by birtles** [comment]
I have a plan to add something similar but not for Anki per se (something much better!). Can you point me to any description of how the other add-ons work or describe how you'd like it to work? For example, do you just want to accumulate words and then export them to a tsv? Or use Ankiweb?

**Comment by birtles** [comment]
Thanks for your kind words too.

**Comment by nonetrix** [comment]
This would be great for now I am just using a terminal app I threw together that just makes a card from text input

**Comment by birtles** [comment]
Thanks for the feedback! We've been working on this for quite a while now but it's still at least a few months off I'm afraid. It's definitely coming though!

**Comment by hunt-g** [comment]
Is this still in progress?

**Comment by birtles** [comment]
> Is this still in progress?

Yes, very much! I still don't have an ETA but we're working on it every day. Thank you for your patience.

**Comment by nonetrix** [comment]
Take your time :3

**Comment by birtles** [comment]
Thank you!

**Comment by hunt-g** [comment]
> > Is this still in progress?
> 
> Yes, very much! I still don't have an ETA but we're working on it every day. Thank you for your patience.

Glad to hear, and thanks for the quick reply. Is there an open pull request or a branch for this feature?

**Comment by nonetrix** [comment]
I looked and I couldn't find one

**Comment by birtles** [comment]
I'm afraid we're working on this in a separate repo which we're keeping under wraps until it's more ready.

**Comment by xrishox** [comment]
is there any update on this? i really love using my ipad to consume content, and the dictionary works great, but not being able to mine is a huge bummer. even a way to save words and export words would be better than nothing.

**Comment by birtles** [comment]
> is there any update on this? i really love using my ipad to consume content, and the dictionary works great, but not being able to mine is a huge bummer. even a way to save words and export words would be better than nothing.

Thanks for your interest in this! I'm working on this full time. It's on the way but it's a pretty big feature and I'm trying to find a way to trim it down so I can release it sooner.

**Comment by nonetrix** [comment]
Take your time, it's fine :3

**Comment by jamesellinger** [comment]
> is there any update on this? i really love using my ipad to consume content, and the dictionary works great, but not being able to mine is a huge bummer. even a way to save words and export words would be better than nothing.

If it helps, you could look at this iOS Shortcut that I made. I've been mining with 10ten and importing in Anki with it for several months now:

[10ten Add Anki Card](https://www.icloud.com/shortcuts/b982b3b5bfba42328b2655e55f647914)

You'll have to look at the comments to figure out what values to change for your decks. iOS shortcuts are pretty easy to use, so hopefull I commented it well enough for others to adapt.

Procedure is to choose your word, then tap the word and choose `Tab-seperated` to copy the information to the clipboard. Then run the Shortcut.

You can also add it to "Favorites" in Safari like shown in the image below. Although I forot how I added it, sorry.

![10ten shortcut](https://github.com/birchill/10ten-ja-reader/assets/28816526/6504a2c8-721a-42ba-b9f7-b0f0a5fc9036)

**Comment by jonathan-s** [comment]
Adding my vote on importing to anki, though it seems like it won't come..

**Comment by birtles** [comment]
Thank you for the feedback! Sorry for the delay. This is definitely in the works but is taking time. I'm currently in the final stages of getting some more full-time help on this project as well as setting up a website where less technical users can provide feedback and provide input on priorities.

**Comment by jonathan-s** [comment]
@birtles I was thinking of a possibility to save entries and then export all of them in a anki friendly format. If I made a PR like that would it be appreciated. After exporting, you could clear the saved entries. A workflow like doesn't seem overly complex to implement and would solve most of what I would be looking for in an "anki integration".

**Comment by nonetrix** [comment]
> > is there any update on this? i really love using my ipad to consume content, and the dictionary works great, but not being able to mine is a huge bummer. even a way to save words and export words would be better than nothing.
> 
> If it helps, you could look at this iOS Shortcut that I made. I've been mining with 10ten and importing in Anki with it for several months now:
> 
> [10ten Add Anki Card](https://www.icloud.com/shortcuts/b982b3b5bfba42328b2655e55f647914)
> 
> You'll have to look at the comments to figure out what values to change for your decks. iOS shortcuts are pretty easy to use, so hopefully I commented it well enough for others to adapt.
> 
> Procedure is to choose your word, then tap the word and choose `Tab-seperated` to copy the information to the clipboard. Then run the Shortcut.
> 
> You can also add it to "Favorites" in Safari like shown in the image below. Although I forot how I added it, sorry.
> 
> ![10ten shortcut](https://github.com/birchill/10ten-ja-reader/assets/28816526/6504a2c8-721a-42ba-b9f7-b0f0a5fc9036)

Maybe someone could try making a simple js user script? Idk if it would be possible, but maybe???

**Comment by birtles** [comment]
> [@birtles](https://github.com/birtles) I was thinking of a possibility to save entries and then export all of them in a anki friendly format. If I made a PR like that would it be appreciated? After exporting, you could clear the saved entries. A workflow like doesn't seem overly complex to implement and would solve most of what I would be looking for in an "anki integration".

That's pretty much what I'm implementing—just in a very very complicated way 😅

**Comment by birtles** [comment]
> As they say, perfect is the enemy of good. Isn't there anyway of reducing scope so this feature can get out of the door and then enhancing it along the way?

I've gone through my list of remaining steps and paired it down the bare essentials. Let's see how we go with that.

**Comment by jonathan-s** [comment]
Looking forward to it.

**Comment by tlcarus** [comment]
It would be great if it had the option to use ankiconnect for this like yomichan does on desktop. I have some experience writing some tools that use ankiconnect (its pretty straightforward to work with) but I'm not familiar enough with the 10ten codebase to know how to implement it here, may give it a shot at some point though...

**Comment by birtles** [comment]
> Minor feedback, I'm hoping that anki integration works standalone on an iOS device in case I'm not home to avoid needing a computer for post-processing or AnkiConnect.

Yes, that's very much the plan. We have a prototype of that but that feature probably won't be part of the initial release.

> Would love to have the actual sentence the word is from. This would go a long way in making flash cards context rich.

This is also definitely the plan. That part should be there from the first version.

> Also it might be a long stretch but would be great to have pitch accent.

That might take a bit longer but it's worth considering. Thanks for raising that.

**Comment by jonathan-s** [comment]
Checking in if it's come a bit closer to release.

**Comment by AlexW00** [comment]
any updates?

**Comment by AlexW00** [comment]
> any updates?

i implemented anki integration similar to how it works on yomitan - https://github.com/birchill/10ten-ja-reader/pull/2766
